### ISSUE TITLE
Problematic recommendation 

### ISSUE THREAD

**Issue by hvr** [minad/intro-prelude#1]
I just noticed `intro-prelude` which recommends to

> Intro reexported as Prelude. Replace base with base-noprelude and intro-prelude in your build-depends. 

IOW, this means that users of `intro-prelude` are recommended to say e.g.

```
build-depends: base-noprelude == 4.9.*, intro-prelude == 0.1.*
```

However, while I designed `base-noprelude` carefully to have a meaningful version number which identifies the exposed API, `intro-prelude` does not have a good version number because currently `intro-prelude` does not constraint its dependencies, i.e.

```
name: intro-prelude
version: 0.1.0.0
library
  build-depends:  intro
  reexported-modules: Intro as Prelude
```

If packages start depending on `intro-prelude` in the way recommended above, we'd run into serious issues on Hackage cause if packages break due to `intro-prelude-0.1.0.0` having a variable API that can't be controlled, I'd have little choice but perform drastic `.cabal` metadata revisions which would have serious undesired side-effects.

long story short: `intro-prelude` ought to make sure that whatever API it reexports is properly identified by its version number, otherwise it will cause problems for everyone involved.

**Comment by minad** [comment]
Cool, thank you for your comment. intro-prelude should probably just copy the version of intro, right?

This thing is a rather quick experiment with base-noprelude which isn't working very well at this point because of issues with ghci, cabal module autogeneration, etc. Maybe in the future this becomes more viable.

**Comment by hvr** [comment]
@minad yes, [editing in a rather tight version bound](https://hackage.haskell.org/package/intro-prelude-0.1.0.0/intro-prelude.cabal/edit) (i.e. at least a minor-version upper bound to make sure the set of exported symbols remains fixed) would be the easiest way, especially since it's quite tedious to have an explicit export list for a `Prelude`-style module (which would allow to use a major-version upper bound; assuming there's no risk that `intro` will define typeclass instances, as we can't control those)
